storage/watchable_store.go

```diff
@@ -287,6 +291,15 @@ func (s *watchableStore) syncWatchers() {
 		s.unsynced.delete(w)
 	}
 
+	// bring all un-notified watchers to synced.
```

Comment:
no. they are different thing. for the unsynced watchers that we do not have events to send, we should directly move them to sync. these lines of code only solves this problem.